File: spark-tools/src/test/java/fr/rozanc/bigdata/spark/TypedDatasetGeneratorTest.java
package fr.rozanc.bigdata.spark;

import fr.rozanc.bigdata.spark.sample.SampleData;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class TypedDatasetGeneratorTest {

    private TypedDatasetGenerator generator;

    @BeforeEach
    void setUp() {
        generator = new TypedDatasetGenerator();
    }

    @Test
    void test() {
        generator.forType(SampleData.class);
    }
}

File: spark-typed-dataset-generator/src/test/java/fr/rozanc/bigdata/spark/generator/TypedDatasetGeneratorTest.kt
package fr.rozanc.bigdata.spark.generator

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.io.PrintWriter

internal class TypedDatasetGeneratorTest {

    @Test
    fun testClassGeneration() {
        val generator = TypedDatasetGenerator()
        generator.write(className = "PersonDataset",
                destinationPackage = "test",
                underlyingType = "Person",
                fieldNames = listOf("aaa", "bbb", "ccc"),
                serialVersionUid = 1L,
                printWriter = PrintWriter(System.out))
    }
}